### PR TITLE
Add docs entry regarding DYLD_LIBRARY_PATH

### LINKED ISSUE #816: Properly set DYLD_LIBRARY_PATH for macOS wheel delocation
I have a Github Action Workflow that builds wheels with native extensions for Linux, macOS, and Windows. External native dependencies for the macOS builds are installed via Vcpkg, so I need to set the  `CPATH` and `LIBRARY_PATH` environment variables to `/usr/local/share/vcpkg/installed/x64-osx/include` and `/usr/local/share/vcpkg/installed/x64-osx/lib` in order for clang to find the headers and `*.dylib` binaries.

The resulting macOS binaries have shared library dependencies starting with `@rpath` placeholders. Since there is not `RPATH` hardcoded in the binaries themselves, `delocate-wheel` needs a properly configured `DYLD_LIBRARY_PATH` for resolving these. Unfortunately, I seem to be unable to set either of `DYLD_LIBRARY_PATH` and `LD_LIBRARY_PATH` on a Workflow job, so this variable is always unset, no matter what (I couldn't find any documentation, but it's probably a built-in precaution). I also tried setting it indirectly via `CIBW_ENVIRONMENT`, but without success.

My workaround is to add a step like that:
```yaml
- name: Delocate Wheels (macOS)
  if: runner.os == 'macOS'
  run: |
    set -e
    python3 -m pip install delocate
    for w in ./wheelhouse/*-macosx*.whl; do
      DYLD_LIBRARY_PATH="$LIBRARY_PATH" python3 -m delocate.cmd.delocate_wheel -v "$w"
    done
```
It would be great if cibuildwheel could take care of that on its own. This issue pertains mainly to macOS, but may apply to Linux as well (not tested, because I install dependencies directly via `yum` there).

### PR DESCRIPTION
**Pull request by joerick**
As discussed in #816.

Fixes #816.

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -279,7 +279,21 @@ python3 -m pip install twine
...
+
+macOS has built-in [System Integrity protections](https://developer.apple.com/library/archive/documentation/Security/Conceptual/System_Integrity_Protection_Guide/RuntimeProtections/RuntimeProtections.html) which limits the use of `DYLD_LIBRARY_PATH` and `LD_LIBRARY_PATH` so that it does not automatically pass to children processes. This means if you set `DYLD_LIBRARY_PATH` before running cibuildwheel, or even set it in `CIBW_ENVIRONMENT`, it will be stripped out of the environment before delocate is called.
+
+To work around this, use a different environment variable such as `REPAIR_LIBRARY_PATH` to store the library path, and set `DYLD_LIBRARY_PATH` in [`CIBW_REPAIR_WHEEL_COMMAND_MACOS`](https://cibuildwheel.readthedocs.io/en/stable/options/#repair-wheel-command), like this:
+
+```yaml
+CIBW_REPAIR_WHEEL_COMMAND_MACOS: >
+  DYLD_LIBRARY_PATH=$REPAIR_LIBRARY_PATH delocate-listdeps {wheel} &&
+  DYLD_LIBRARY_PATH=$REPAIR_LIBRARY_PATH delocate-wheel --require-archs {delocate_archs} -w {dest_dir} {wheel}
+```
```
Comment (henryiii) [inline docs/faq.md:292]: Can we do tabs and show the TOML version?

Comment (joerick) [reply docs/faq.md:292]: yeah, that would be nice

Comment (joerick) [reply docs/faq.md]: I've added them. Look okay to you?

File: docs/faq.md [hunk 2]
```diff
@@ -279,7 +279,33 @@ python3 -m pip install twine
...
+    CIBW_REPAIR_WHEEL_COMMAND_MACOS: >
+        DYLD_LIBRARY_PATH=$REPAIR_LIBRARY_PATH delocate-listdeps {wheel} &&
+        DYLD_LIBRARY_PATH=$REPAIR_LIBRARY_PATH delocate-wheel --require-archs {delocate_archs} -w {dest_dir} {wheel}
+    ```
+
+!!! tab examples "pyproject.toml"
+
+    ```toml
+    [tool.cibuildwheel.macos]
+    repair_wheel_command = [
```
Comment (henryiii) [inline docs/faq.md:300]:
```suggestion
    repair-wheel-command = [
```

Comment (joerick) [reply docs/faq.md]: good catch! thanks

**Review by henryiii — APPROVED** [review]

**Pull request merged**